src/main/java/seedu/address/model/person/Grade.java

```diff
@@ -0,0 +1,42 @@
+package seedu.address.model.person;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class Grade {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Grades should only contain an alphabet, and it should not be blank";
+
+    public HashMap<String, double[]> marks;
+
+    /**
+     * Constructs a {@code Grade}.
+     */
+    public Grade() {
+        marks = new HashMap<>();
+    }
+
+    public double totalMarks(HashMap<String, double[]> subjectMarks) {
+        ArrayList<double[]> totalMarksArray = new ArrayList<>();
+        totalMarksArray.addAll(subjectMarks.values());
+        int length = totalMarksArray.size();
+        double totalMarks = 0;
+        for (int i = 0; i < length; i++) {
+            totalMarks = totalMarksArray.get(i)[0] * totalMarksArray.get(i)[1];
+        }
+        return totalMarks;
+    }
```

Comment:
Add a JavaDoc please. To my understanding, this gets the percentage of the total grade of the subject that the student has obtained so far? Perhaps we should normalise this value instead to get the grade that the student would obtain at the moment, e.g. of the student got 40 out of a possible 50% so far, we could return 80% (for this method)

Also, not sure if line 26 should have "+=" instead of "=".